lib/jsdom/living/nodes/Element-impl.js

```diff
@@ -430,6 +430,43 @@ class ElementImpl extends NodeImpl {
 
     return attr;
   }
+
+  insertAdjacentHTML(position, text) {
+    const parentNode = this.parentNode;
+    const container = this.ownerDocument.createElementNS("http://www.w3.org/1999/xhtml", "_");
```

Comment:
You should use `this.ownerDocument.createElement("template")` as the container. Also you should add some tests showing how using `"_"` fails for elements like `thead`, `tr`, `td`.